tests/e2e/options.go

```diff
@@ -107,26 +199,54 @@ func DefaultAgentOpt() *AgentOpt {
 	return opt
 }
 
+// DefaultPingOpts creates default values for PingOpt
+func DefaultPingOpts() *PingOpt {
+	return &PingOpt{
+		AllowedLoss: 49, // by default at least half of the packets should get through
+		MaxTimeout:  4,
+	}
+}
+
 // WithoutVPPAgent is test setup option disabling vpp-agent setup
 func WithoutVPPAgent() SetupOptModifier {
 	return func(o *SetupOpt) {
 		o.SetupAgent = false
 	}
 }
 
-// WithEtcd is test setup option enabling vpp-agent setup
+// WithCustomVPPAgent is test setup option using alternative vpp-agent image (customized original vpp-agent)
+func WithCustomVPPAgent() SetupOptModifier {
+	return func(o *SetupOpt) {
+		o.AgentOpt.Image = "vppagent.test.ligato.io:custom"
```

Comment:
Shouldn't this option accept parameter to allow defining truly _custom_ image?

Reply:
I'm using the [customized vpp-agent image made in e2e script](https://github.com/ligato/vpp-agent/pull/1779/files#diff-f55dfa5c52801afc2b7175e0e2343b181f02bd2329837e80e769f2498c213cd7R44-R48). This allows me to prepare this image only once per run of all e2e tests. If i make this method truly custom by introducing options, then i will have to prepare the image in test runtime. That means image creation per test and not per one test suite run. If 2 test want to use the custom image(doesn't happen now, but could in future), they do not know whether it was rebuilded using the latest vpp-agent image as base image or not. So it gets rebuilded multiple times. So i choose the easy way to rebuild it only once before test suite run, but the downside is that it is hardcoded.